### DISCUSSION TITLE
Has anyone implemented a Facebook/Instagram like activity feed with RavenDB?

### DISCUSSION THREAD

**Discussion by pmengal** [ravendb/ravendb#14211] [Q&A] [answered]
Hi,

We have multiple objects stored in RavenDB that are domain events. We would like to give users the possibility to follow other users to see their activity (on any of those event objects) in  their timeline.

Has anyone implemented with success this use case with RavenDB?

Secondary question: how would you store follows? In a collection on the user object or in a distinct collection.

Found this [conversation](https://groups.google.com/g/ravendb/c/dioENla2dEw/m/TGbIkt26AgAJ) in the archive , but someone might come with a better solution

Thanks in advance

Pierre

**Comment by ayende** [comment]
A huge aspect of this is what scale you need it. If we are talking about something like a departmental feed, where you have well defined scope, that is one thing ( `User.Following` array that you just store all those you follow).

If you are talking about a social network with may have 100K+ people following the same person, that is something quite different. 

Can you provide some more context on what you need?

**Comment by ayende** [comment, marked as answer]
There are quite a few resources that I can point you to. For example, to do something the size of Twitter, I wrote a 10 blog posts series on the topic here: https://ayende.com/blog/posts/series/193026-C/building-a-social-media-platform-without-going-bankrupt

The simplest approach is described here: https://ayende.com/blog/194401-A/modeling-social-network-connection-with-ravendb

**Reply by pmengal** [reply]
How could I combine this with a multimap index that would map each even object to an ActivityFeedResult object that would contain the timestamp the ID of the followed user, the type of event and it's RavenDB ID? For a given user, I could create the feed based on who he is following very easily and query the index.

**Reply by ayende** [reply]
Go back one stage, please. 
What are the sizes involved? How many people can you follow? What is the expected number of posts / activities you'll track.

In general, the easiest way to do that is to just query like this:

`from index AllActivities where AuthorID in ($followingIds) order by Date desc limit 50`

**Reply by pmengal** [reply]
I have a test dataset with 350K users and 6M events for the past 6 months. In "best" cases scenarios, this will never be more than 10 times that. Common case will be around 1M users and few millions events. It's very likely that top users will be followed by a large proportion of others.

**Reply by ayende** [reply]
The actual question is more interesting, however. How many people do you expect a single user to follow? (I assume that this is the key one, no?)

If this is a test db, I would love to lay my hands on it as well as your queries, we can work on that together.

**Reply by pmengal** [reply]
It's in the eSports field. So if we take the soccer analogy, the probability that a high number of members would follow Cristiano Ronaldo would be very high. Also, there is a lot of famous players like this.

I send you a separate email with accesses to the development database in a minute.

**Reply by ayende** [reply]
Yes, high number of followers is expected. The question is the other way around, how many players is a single user following?

I assume that the number is both finite and small.

For example, on Twitter, there is a hard limit of 5000 users you can follow. Here is how this looks like:

![image](https://user-images.githubusercontent.com/116915/169653279-b47fdbc9-01b6-4f89-82ef-284ee530e4e0.png)


In other words, the number I care about for this query is the `Following`, not `Followers`.
